src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4954,13 +4961,25 @@ public override bool PreVisit(CallNode node)
                         _txb.AddVolatileVariables(args[i], volatileVariables);
                     }
 
+                    var isIdentifier =
+                        args[i] is FirstNameNode &&
+                        _features.HasFlag(Features.SupportIdentifiers) &&
+                        maybeFunc.IsIdentifierParam(i);
+
                     // Use the new scope only for lambda args.
-                    _currentScope = (maybeFunc.IsLambdaParam(i) && scopeInfo.AppliesToArgument(i)) ? scopeNew : scopeNew.Parent;
-                    args[i].Accept(this);
+                    _currentScope = ((maybeFunc.IsLambdaParam(i) || isIdentifier) && scopeInfo.AppliesToArgument(i)) ? scopeNew : scopeNew.Parent;
```

Comment:
>isIdentifier [](http://example.com/codeflow?start=68&length=12)

does isIdentifier need to affect the scope here? Seems mostly irrelevant.

Reply:
I don't know if we should or should not use isIdentifier here but tests show it's not necessary and I've removed it.